image-download

```diff
@@ -68,20 +73,21 @@ def find(name, stores, latest, quiet):
             ai = socket.getaddrinfo(url.hostname, url.port or defport, socket.AF_INET, 0, socket.IPPROTO_TCP)
         except socket.gaierror:
             ai = []
-            message = store
 
         for (family, socktype, proto, canonname, sockaddr) in ai:
-            message = "{scheme}://{0}:{1}{path}".format(*sockaddr, scheme=url.scheme, path=url.path)
-
             def curl(*args):
                 try:
                     cmd = ["curl"] + list(args) + ["--head", "--connect-timeout", "10", "--silent",
                                                    "--fail", "--cacert", ca, source]
                     start = time.time()
                     output = subprocess.check_output(cmd, universal_newlines=True)
-                    found.append((cmd, output, message, time.time() - start))
+                    duration = time.time() - start
+                    found.append((cmd, output, store, duration))
                     if not quiet:
-                        sys.stderr.write(" > {0}\n".format(message))
+                        sys.stderr.write("{0} {1} [{2}ms]\n".format(
+                            SYMBOLS['present'],
+                            store,
+                            int(duration * 1000 + 0.5)))
```

Comment:
This addition is pretty nice, actually.  It gives a bit of a hint as to the criteria in use.